nilearn/interfaces/fmriprep/load_confounds_utils.py

```diff
@@ -255,20 +283,26 @@ def get_confounds_file(image_file, flag_full_aroma):
     flag_full_aroma : :obj:`bool`
         True if the input is a full ICA-AROMA output, False otherwise.
 
+    flag_tedata : :obj:`bool`
+        True if the input is a TEDANA optimally combined output, False otherwise.
+
     Returns
     -------
     confounds_raw_path : :obj:`str`
         Path to the associated confounds file.
     """
-    _check_images(image_file, flag_full_aroma)
-    confounds_raw_path = _get_file_name(image_file)
+    _check_images(image_file, flag_full_aroma, flag_tedana)
+    confounds_raw_path = _get_file_name(image_file, flag_tedana=flag_tedana)
     return confounds_raw_path
 
-
-def get_json(confounds_raw_path):
+def get_json(confounds_raw_path, flag_tedana=False):
     """Return json data companion file to the confounds tsv file."""
-    # Load JSON file
-    return str(confounds_raw_path).replace("tsv", "json")
+    if flag_tedana:
+        # TEDANA does not have a json confound companion file
+        return None
+    else:
+        # Load JSON file
+        return str(confounds_raw_path).replace("tsv", "json")
```

Comment:
we return early
else not needed

```suggestion
    # Load JSON file
    return str(confounds_raw_path).replace("tsv", "json")
```